apps/server/src/controllers/userControllers/getUserTransactionHistory.ts

```diff
@@ -1,16 +1,47 @@
 import { Response } from 'express'
+import { Op } from 'sequelize'
 import { RequestExt } from '../../middleware/authorization/authentication'
 import Transactions from '../../models/transactions'
 import { HTTP_STATUS_CODE } from '../../constants'
+import Wallets from '../../models/wallets'
+import Users from '../../models/users'
 
 export const getTransactionHistory = async (req: RequestExt, res: Response) => {
   try {
     const { _userId: userId } = req.body
-    const transactionHistory = await Transactions.findAll({
-      where: { ownerId: userId },
+
+    const startDate = new Date()
+    startDate.setHours(0, 0, 0, 0) // Set hours to beginning of the day
+    startDate.setDate(startDate.getDate() - startDate.getDay()) // Set date to Sunday
+
+    // Get the end date of the current week (Saturday)
+    const endDate = new Date(startDate)
+    endDate.setDate(startDate.getDate() + 6) // Add 6 days to get to Saturday
+
+    const userTransactions = await Transactions.findAll({
+      where: {
+        ownerId: userId,
+        // Query the database to get data within the current week
+        createdAt: { [Op.between]: [startDate, endDate] },
+      },
       order: [['createdAt', 'DESC']],
-      limit: 5,
+      include: [{ model: Wallets }, { model: Users }],
     })
+
+    // Extract relevant information for the response
+    const transactionHistory = userTransactions.map((transaction) => ({
```

Comment:
Let's paginate this at once

Ask @Don-christo  for help here:

or reference https://github.com/Ajo-Savings-Project/ajo-savings/blob/7243c8977e8979485ca7557397197cb7311d213e/apps/server/src/controllers/userControllers/getUpcomingActivities.ts#L103